File: day3/solution_test_python/tests/test_my_file.py
# File: tests/test_my_file.py

from my_file import calculate_incidence, classify_risk

def test_calculate_incidence():
    assert calculate_incidence(5, 1e6, 'area51') == 0.5
    assert calculate_incidence(0, 1, 'area51') == 0
    assert calculate_incidence(0, 0, 'area51') is None

def test_classify_risk():
    assert classify_risk(None, 'area51') == "unknown"
    assert classify_risk(5,'area51') == "low"
    assert classify_risk(50, 'area51') == "low"
    assert classify_risk(100, 'area51') == "high" 

File: day3/solution_test_R/tests/testthat/test-my_file.R
source("../../my_file.R")

test_that("calculate_incidence works", {
  expect_equal(calculate_incidence(5, 1e6), 0.5)
  expect_equal(calculate_incidence(0, 1), 0)
  expect_warning(result <- calculate_incidence(0, 0), "Population is zero. Incidence rate cannot be calculated.")
  expect_true(is.na(result))
})

test_that("classify_risk works", {
  expect_equal(classify_risk(NA_real_), NA)
  expect_equal(classify_risk(5), "low")
  expect_equal(classify_risk(50), "low")
  expect_equal(classify_risk(100), "high")
})
